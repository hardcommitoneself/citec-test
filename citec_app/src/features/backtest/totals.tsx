// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { Icon } from 'assets/icons'
import CardStandard from 'features/ui/cards/card-standard'
import { BacktestRes } from 'features/utils/interfaces/backtest-api-resp'

type TotalsProps = BacktestRes

export const Totals = ({
   capital_gains,
   tracking_expected,
   fundamentals,
}: TotalsProps) => {
   const nf = new Intl.NumberFormat('en-US')

   const { unrealized_cap_losses, realized_cap_gains, outstanding_balance } =
      capital_gains
   const { expected_tracking_kpis } = tracking_expected
   const { total_positions } = fundamentals

   const calculatePercentaje = (value: number) => {
      if (value === 0) return '0'

      return (value * 100) / outstanding_balance
   }

   const cards = [
      {
         title: 'Expected tracking error',
         value: `${(expected_tracking_kpis.tracking_error * 100).toFixed(3)} %`,
         icon: <Icon.Error />,
      },
      {
         title: 'Unrealized losses to harvest',
         value: `$ ${nf.format(unrealized_cap_losses)}`,
         percentageBalance: calculatePercentaje(unrealized_cap_losses),
         numberPositions: total_positions,
         icon: <Icon.Harvest />,
      },
      {
         title: 'Realized capital gains',
         value: `$ ${nf.format(realized_cap_gains)}`,
         percentageBalance: calculatePercentaje(realized_cap_gains),
         numberPositions: total_positions,
         icon: <Icon.Capital />,
      },
   ]

   return (
      <div className="grid grid-cols-3 gap-x-11 w-full">
         {cards.map(
            ({ title, value, icon, numberPositions, percentageBalance }, i) => (
               <CardStandard key={i} className="!px-3 !py-5 flex flex-col">
                  <div className="flex flex-row justify-between relative ">
                     <h3 className="font-bold text-neutral-600 text-sm">
                        {title}
                     </h3>

                     {icon}
                  </div>

                  <span className="text-neutral-600 font-semibold text-2xl">
                     {value}
                  </span>
                  {percentageBalance && (
                     <span className="text-neutral-500 text-sm mt-2">
                        % of outstandig balance: {percentageBalance} %
                     </span>
                  )}

                  {numberPositions && (
                     <span className="text-neutral-500 text-sm mt-2">
                        Number of positions: {numberPositions}
                     </span>
                  )}
               </CardStandard>
            )
         )}
      </div>
   )
}
