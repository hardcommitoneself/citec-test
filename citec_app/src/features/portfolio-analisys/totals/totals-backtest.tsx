// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { renderPopup } from 'features/popups/backtest-popups'
import { TotalsBox } from 'features/ui/cards'
import { BacktestRes } from 'features/utils/interfaces/backtest-api-resp'
import { numberToFixed } from 'features/utils/to-fixed-number'

const _tracking = require('assets/img/tracking_grey.png')
const equal = require('assets/img/equal.png')
const not = require('assets/img/not.png')

interface TotalBacktestProps {
   backtest: BacktestRes
}

export const TotalBacktest = ({ backtest }: TotalBacktestProps) => {
   const { risk, tracking } = backtest

   return (
      <div className="grid grid-cols-3 gap-6 mt-2">
         <TotalsBox
            value={numberToFixed(tracking.tracking_difference * 100) + ' %'}
            text="Tracking Difference"
            img={_tracking}
            functInfo={() => renderPopup('trackDiffAlert')}
            info
         />
         <TotalsBox
            value={numberToFixed(risk.beta)}
            text="Beta"
            img={not}
            functInfo={() => renderPopup('betAlert')}
            info
         />
         <TotalsBox
            value={numberToFixed(tracking.r_square)}
            text="R-Square"
            img={equal}
            functInfo={() => renderPopup('rsqrAlert')}
            info
         />
      </div>
   )
}
