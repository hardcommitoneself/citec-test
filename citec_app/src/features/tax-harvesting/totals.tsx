// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { TotalsBox } from 'features/ui/cards'
import { formatToMoney } from 'features/utils/format-to-money'
import { useAppSelector } from 'store/hooks'

export const TotalsTaxHarvesting = () => {
   const taxHarvestingState = useAppSelector((state) => state.taxHarvesting)

   const {
      arrayPositionSelected,
      unrealizedCapitalGains,
      unrealizedCapitalGainsNegative,
   } = taxHarvestingState

   const negativePositionsSelected = arrayPositionSelected
      .map((values) => values.unrealized_capital_gains)
      .filter((values) => values < 0)

   let totalNegativesPositionsSelected = 0

   if (negativePositionsSelected.length > 0) {
      totalNegativesPositionsSelected = negativePositionsSelected.reduce(
         (prev, curr) => prev + curr
      )
   }

   const unrealizedCapitalGainsNumber = formatToMoney(unrealizedCapitalGains)

   const totalHarvestinOpportunity =
      unrealizedCapitalGainsNegative - totalNegativesPositionsSelected

   const harvestingOpportunityNumber = formatToMoney(totalHarvestinOpportunity)

   const lossesHarvestedNumber = formatToMoney(totalNegativesPositionsSelected)

   const cards = [
      {
         img: require('assets/img/bolsa.png'),

         title: ' Unrealized Gains',
         value: unrealizedCapitalGainsNumber,
      },
      {
         img: require('assets/img/espiga.png'),
         title: 'Harvesting Opportunity',
         value: harvestingOpportunityNumber,
      },
      {
         img: require('assets/img/espiga_gray.png'),
         title: 'Losses Harvested',
         value: lossesHarvestedNumber,
      },
   ]

   return (
      <div className="w-full grid grid-cols-3 gap-11 ">
         {cards.map(({ title, value, img }, i) => (
            <TotalsBox key={i} img={img} value={value} text={title} bgWhite />
         ))}
      </div>
   )
}
