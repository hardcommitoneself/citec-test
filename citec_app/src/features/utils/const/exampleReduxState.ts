// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

export const exampleReduxState = {
   errors: '',
   taxHarvesting: {
      arrayPositionSelected: [
         {
            outstanding_balance: 0,
            ticker: '$CASH',
            unrealized_capital_gains: 500,
            taxAlpha: -200,
         },
      ],
      positionsTable: [
         {
            outstanding_balance: 50000,
            ticker: 'AAPL',
            unrealized_capital_gains: -1000,
            taxAlpha: 400,
         },
         {
            outstanding_balance: 0,
            ticker: '$CASH',
            unrealized_capital_gains: 500,
            taxAlpha: -200,
         },
      ],
      taxAlpha: -400,
      unrealizedCapitalGains: -500,
      unrealizedCapitalGainsNegative: -1000,
      unrealizedCapitalGainsSelected: 0,
   },
   portfolio: {
      ticker: ['$CASH', 'AAPL'],
      outstanding_balance: [1000000, 50000],
      unrealized_capital_gains: [0, 11234],
      analysis: {
         fundamentals: {
            total_assets: 2,
            total_positions: 2,
            pe: 24.32,
            pb: 38.96,
            dividend_yield: 0.59,
         },
         risk: {
            sharpe: 0.305683,
            volatility: 0.0147,
            beta: 0.048719,
            alpha: 0.000035,
         },
         returns: {
            returns_1M: -0.0021,
            returns_3M: -0.0037,
            returns_6M: -0.0028,
            CAGR: -0.1058,
         },
         capital_gains: {
            outstanding_balance: 1050000,
            unrealized_cap_losses: 0,
            unrealized_cap_gains: 11234,
            realized_cap_losses: 0,
            realized_cap_gains: 0,
            taxes_cost: 0,
            taxes_rate: 0.4,
         },
         sectors: {
            sector: ['Cash', 'Information Technology'],
            percentage: [0.952381, 0.047619],
            percentage_benchmark: [0.3333333333333333, 0.6666666666666666],
         },
         factors: {
            factor: ['style', 'size'],
            percentage: [0.502991, 0.52381],
         },
         segment_matrix: {
            style: ['Val', 'Val'],
            size: ['Large', 'Mid'],
            weight: [0, 0],
         },
         top_holdings: {
            ticker: ['$CASH', 'AAPL'],
            weight: [0.952381, 0.047619],
            outstanding_balance: [1000000, 50000],
            unrealized_capital_gains: [0, 11234],
         },
         concentration: {
            ticker: ['$CASH', 'AAPL'],
            concentration: [95.238095, 100],
         },
         var: {
            quantile: [2, 4],
            value: [0.000072, 0.001334],
         },
         risk_returns: {
            ticker: ['portfolio', 'benchmark'],
            returns: [0.004722, 0.119508],
            volatility: [0.008561, 0.205196],
            sharpe: [0.551568, 0.582407],
         },
      },
   },
   universe: {
      factors_filter: [
         {
            factor: 'style',
            ranges: [0, 100],
         },
         {
            factor: 'size',
            ranges: [0, 100],
         },
         {
            factor: 'yield',
            ranges: [0, 100],
         },
      ],
      sectors_filter: [
         'Industrials',
         'Financials',
         'Health Care',
         'Consumer Staples',
         'Communication Services',
         'Information Technology',
         'Consumer Discretionary',
         'Materials',
         'Cash',
         'Real Estate',
         'Energy',
         'Utilities',
      ],
      to_exclude: [],
      to_include: [],
      universe_base: 'SPDR S&P 500',
   },
   rebalance: {
      ticker: ['$CASH', 'AAPL'],
      hold: [0, 1],
      objective: {
         name: 'Minimize tracking error',
         key: 'min_tracking_error',
      },
      benchmark: 'SPY',
      min_weight: 0.000001,
      number_stocks: [10, 200],
   },
   portfolio_optimized: {
      portfolio_id: null,
      portfolio: 'optimized_portfolio',
      benchmark: 'SPY',
      creation_date: null,
      ticker: ['AAPL', 'MSFT'],
      outstanding_balance: [72506.52, 63477.37],
      unrealized_capital_gains: [0, 0],
      optimization_status: 'Optimization terminated successfully',
   },
   trades: [
      {
         ticker: 'MRK',
         operation: 'SELL',
         volume: 275161.76,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 3053,
         expected_volume: 275193,
         delta_volume: 31,
         price: 90.1388,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/2eedcaa6-80ec-11ea-853f-00000000092a.png',
            name: 'Merck & Co Inc',
         },
         gain_loss: [0, 0],
      },
      {
         ticker: 'BMY',
         operation: 'SELL',
         volume: 152166.26,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 2072,
         expected_volume: 152146,
         delta_volume: -19,
         price: 73.43,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/8b628318-80eb-11ea-93a2-00000000092a.png',
            name: 'Bristol-Myers Squibb Co',
         },
         gain_loss: [0, 0],
      },
      {
         ticker: 'WMT',
         operation: 'SELL',
         volume: 83898.27,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 679,
         expected_volume: 83845,
         delta_volume: -53,
         price: 123.4832,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/52057505dd05a752d364065fcddf7772a700b31c47c39e9dcb41e393f6693d52.png',
            name: 'Walmart Inc',
         },
         gain_loss: [0, 0],
      },
   ],
   workflow: {
      type: 'load',
   },
   static_data: {
      universe: {
         universes: [
            {
               name: 'INVESCO QQQ TRUST SERIES 1',
               description: 'INVESCO QQQ TRUST SERIES 1',
            },
            {
               name: 'ISHARES MSCI USA ESG SELECT',
               description: 'ISHARES MSCI USA ESG SELECT',
            },
         ],
         sectors: [
            'Cash',
            'Communication Services',
            'Consumer Discretionary',
            'Consumer Staples',
         ],
         factors: ['style', 'size', 'yield', 'quality'],
      },
      rebalance: {
         objectives: [
            {
               key: 'min_tracking_error',
               name: 'Minimize tracking error',
            },
            {
               key: 'min_risk_parity',
               name: 'Minimize risk parity',
            },
            { key: 'min_volatility', name: 'Minimize volatility' },
            {
               key: 'max_diversification_ratio',
               name: 'Maximize diversification ratio',
            },
         ],
         benchmarks: [
            {
               key: 'sp_500',
               description: 'S&P 500 universe',
               ticker: 'SPY',
            },
            {
               key: 'nasdaq_100',
               description: 'NASDAQ 100',
               ticker: 'QQQ',
            },
         ],
      },
   },
   brokers: {
      brokerConnected: '',
      balance: 0,
      accountType: [],
      a1c_t: '',
   },
   ideas: [
      {
         portfolio_id: 16,
         portfolio: 'my_first_portfolio_sadasd',
         benchmark: 'SPY',
         creation_date: '2022-04-06 16:10:19',
         ticker: ['MSFT', 'AAPL'],
         outstanding_balance: [200000, 100000],
         unrealized_capital_gains: [0, 35700],
         optimization_status: null,
      },
      {
         portfolio_id: 18,
         portfolio: 'testing',
         benchmark: 'SPY',
         creation_date: '2022-04-06 16:10:19',
         ticker: ['MSFT', 'AAPL'],
         outstanding_balance: [200000, 100000],
         unrealized_capital_gains: [0, 35700],
         optimization_status: null,
      },
      {
         portfolio_id: 19,
         portfolio: 'Jueves',
         benchmark: 'SPY',
         creation_date: '2022-06-23 10:38:00',
         ticker: [
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'AAPL',
            'MMM',
            'TMO',
            'BLK',
            'MSFT',
            'QCOM',
            'NVDA',
            'CSX',
            'ADBE',
            'XOM',
            'AVGO',
            'BAC',
            'TSLA',
            'SHW',
            'TXN',
            'NFLX',
            'GOOGL',
            'C',
            'MMC',
            'BKNG',
            'COST',
            'NKE',
            'DIS',
            'EL',
            'CSCO',
            'BRK.B',
            'META',
            'AMZN',
            'DG',
            'CCI',
            'ORCL',
            'USB',
            'TGT',
            'PG',
            'ICE',
            'GOOG',
            'AMD',
            'MCD',
            'CVS',
            'HD',
            'CME',
            'MU',
            'UNH',
            'ATVI',
            'V',
            'EOG',
            'GS',
            'HUM',
            'COP',
         ],
         outstanding_balance: [
            80352.37, 74072.31, 59343.2, 54736.44, 41924.05, 41510.68, 35441.43,
            27349.74, 26308.48, 26136.99, 25646.73, 25479.44, 24726.53, 24075.27,
            23856.18, 23763.38, 23735.12, 23092.02, 22995.95, 21888.81, 21775.5,
            21090.31, 16734.94, 16192.87, 15720.33, 14020.04, 14018.61, 13225.5,
            12996.74, 11693.52, 11315.64, 10798.53, 10583.72, 10309.54, 10245.54,
            9611.94, 8747.21, 8723.18, 6339.86, 5866.02, 5426.94, 4912.02, 4857.23,
            3915.03, 3395.13, 3370.41, 3322.34, 3184.94, 3001.78, 1983.75, 1858.2,
            1789.12, 1309.98, 1228.49,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0,
         ],
         optimization_status: null,
      },
      {
         portfolio_id: 20,
         portfolio: 'Min volatility',
         benchmark: 'SPY',
         creation_date: '2022-06-23 11:25:41',
         ticker: [
            'MRK',
            'BMY',
            'WMT',
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'MMM',
            'CSX',
            'XOM',
            'SHW',
            'TXN',
            'C',
            'BRK.B',
            'BA',
            'BSX',
            'CB',
            'CI',
            'CL',
            'CME',
            'COP',
            'COST',
            'CRM',
            'ZTS',
            'ADBE',
            'AMZN',
            'AVGO',
         ],
         outstanding_balance: [
            275161.76, 152166.26, 83898.27, 80352.37, 74072.31, 59343.2, 54736.44,
            41924.05, 35441.43, 24726.53, 23856.18, 22995.95, 21888.81, 16734.94,
            11315.64, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49,
            2116.49, 2116.49, 2116.49, 119.48, 96.79, 4.71,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0,
         ],
         optimization_status: null,
      },
      {
         portfolio_id: 21,
         portfolio: 'min2',
         benchmark: 'SPY',
         creation_date: '2022-06-23 11:27:32',
         ticker: [
            'MRK',
            'BMY',
            'WMT',
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'MMM',
            'CSX',
            'XOM',
            'SHW',
            'TXN',
            'C',
            'BRK.B',
            'BA',
            'BSX',
            'CB',
            'CI',
            'CL',
            'CME',
            'COP',
            'COST',
            'CRM',
            'ZTS',
            'ADBE',
            'AMZN',
            'AVGO',
         ],
         outstanding_balance: [
            275161.76, 152166.26, 83898.27, 80352.37, 74072.31, 59343.2, 54736.44,
            41924.05, 35441.43, 24726.53, 23856.18, 22995.95, 21888.81, 16734.94,
            11315.64, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49,
            2116.49, 2116.49, 2116.49, 119.48, 96.79, 4.71,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0,
         ],
         optimization_status: null,
      },
   ],
}
