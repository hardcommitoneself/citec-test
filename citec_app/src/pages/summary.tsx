// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { Button, Tooltip } from '@mui/material'
import { useState } from 'react'
import Swal from 'sweetalert2'

import { AppNavigate } from 'routes'
import { Loading } from 'features/ui'
import { PageContainer } from 'features/ui/page-container'
import {
   useAppNavigate,
   useGetSummary,
   useValidationCriteriaSelected,
} from 'features/utils/hooks'
import { useAppSelector } from 'store/hooks'
import { Backtest, Overview, Risk, Trades } from 'features/summary/tabs'
import { savePortfolio } from 'features/utils/save-portfolio'
import { renderPopup } from 'features/popups/backtest-popups'

type ActiveViews = 'trades' | 'overview' | 'risk' | 'backtest'

const Group = require('assets/img/group.png')

export const Summary = () => {
   const { isLoading, backtest, analysis } = useGetSummary()

   const error = useAppSelector((state) => state.errors)
   const { ticker, outstanding_balance, unrealized_capital_gains } = useAppSelector(
      (state) => state.portfolio_optimized
   )
   const benchmark = useAppSelector((state) => state.rebalance.benchmark)

   const [activeView, setActiveView] = useState<ActiveViews>('trades')

   const navigate = useAppNavigate()
   const views = ['Trades', 'Overview', 'Risk', 'Backtest']
   const disabledViews = ['']

   const isCriteriaSelected = useValidationCriteriaSelected()
   if (!isCriteriaSelected) {
      Swal.fire({
         icon: 'error',
         title: 'Oops...',
         text: 'You need to upload select a benchmark and objective',
      })

      return <AppNavigate to="/criteria" />
   }

   const handleSavePortfolio = () => {
      savePortfolio({
         navigate,
         benchmark,
         outstanding_balance,
         ticker,
         unrealized_capital_gains,
      })
   }

   if (isLoading) return <Loading />

   return (
      <PageContainer error={error}>
         <div className="dashboard_wrapper">
            <div className="InvestmentBody summary dashboard_body">
               <div className="inner_dashboard_cri mt-0">
                  <h1>
                     <span>Congrats!</span> You have created a portfolio. Check if
                     you like it
                  </h1>

                  <div className="options_summary_buttons">
                     {views.map((view) => (
                        <button
                           key={view}
                           className={
                              activeView === view.toLowerCase()
                                 ? 'active-btn'
                                 : 'null'
                           }
                           disabled={disabledViews.includes(view)}
                           onClick={() => {
                              setActiveView(view.toLowerCase() as ActiveViews)
                           }}
                           data-testid={`btn-${view.toLowerCase}`}
                        >
                           {view}
                        </button>
                     ))}
                  </div>
                  <div className="bg-white p-2" data-testid={`active-${activeView}`}>
                     <div className="summary_content_wrapper">
                        { activeView !== 'trades' && (
                           <div className="action_summary">
                              {activeView === 'backtest' && (
                                 <>
                                    <h1>Performance Backtest</h1>
                                    <img
                                       className="moreinfo"
                                       src={Group}
                                       alt=""
                                       onClick={() => {
                                          renderPopup('perfBacktestAlert')
                                       }}
                                       style={{ marginLeft: '15px' }}
                                       data-testid="info-summary-backtest"
                                    />
                                 </>
                              )}
                              {activeView === 'overview' && <h1>Overview</h1>}
                              {activeView === 'risk' && <h1>Risk and Return</h1>}
                              <div className='button_download_trades'>
                                 <Button
                                    className="h-[35px] min-w-0 rounded"
                                    variant="save"
                                    onClick={handleSavePortfolio}
                                 >
                                    <span>Save</span>
                                 </Button>
                              </div>
                           </div>
                        )}
                        {activeView === 'trades' && <Trades />}
                        {activeView === 'backtest' && !!backtest && (
                           <Backtest {...backtest} />
                        )}
                        {activeView == 'overview' && (
                           <Overview backtest={backtest} analysis={analysis} />
                        )}
                        {activeView == 'risk' && <Risk analysis={analysis} />}
                     </div>
                  </div>

               </div>
            </div>
         </div>
      </PageContainer>
   )
}
