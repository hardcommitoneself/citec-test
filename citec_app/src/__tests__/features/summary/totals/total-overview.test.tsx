// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { BrowserRouter } from 'react-router-dom'
import { exampleReduxState } from 'features/utils/const/exampleReduxState'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { TotalOverview } from 'features/summary/totals'

jest.mock('react-apexcharts', () => {
   return {
      __esModule: true,
      default: () => {
         return <div />
      },
   }
})

const backtest = {
   fundamentals: {
      total_assets: 75,
      total_positions: 75,
      pe: 31.733708,
      pb: 11.404961,
      dividend_yield: 1.970252,
   },
   risk: {
      sharpe: -1.107185,
      volatility: 0.239029,
      beta: 1.034119,
      alpha: -1e-5,
   },
   returns: {
      returns_1M: -0.0164,
      returns_3M: -0.0522,
      returns_6M: -0.136,
      CAGR: -0.2258,
   },
   tracking: {
      tracking_error: 0.015409,
      tracking_difference: -4.3e-5,
      information_ratio: -0.710898,
      r_square: 0.99693,
      alpha: -1e-5,
      beta: 1.034119,
   },
   capital_gains: {
      outstanding_balance: 100.01,
      unrealized_cap_losses: 0.0,
      unrealized_cap_gains: 0.0,
      realized_cap_losses: 0.0,
      realized_cap_gains: 0.0,
      taxes_cost: 0.0,
      taxes_rate: 0.4,
   },
   sectors: {
      sector: ['Communication Services', 'Consumer Discretionary'],
      percentage: [0.093291, 0.10149, 0.047595],
   },
   factors: {
      factor: ['style', 'size'],
      percentage: [
         0.527408, 0.986136, 0.411154, 0.638232, 0.275876, 0.597158, 0.423822,
      ],
   },
   segment_matrix: {
      style: ['Val', 'Val', 'Val', 'Bld', 'Bld', 'Bld', 'Gwth', 'Gwth', 'Gwth'],
      size: ['Large', 'Mid', 'Small'],
      weight: [31.526847, 0.0, 0.0],
   },
   tracking_expected: {
      expected_tracking_kpis: {
         tracking_error: 0.015409,
         tracking_difference: -4.3e-5,
         information_ratio: -0.710898,
         r_square: 0.99693,
         alpha: -1e-5,
         beta: 1.034119,
      },
      observation: [0, 1, 2],
      tracking_error: [-0.03003, -0.028923, -0.027815],
      probability: [0.001, 0.0, 0.001],
   },
   ts_backtest: {
      date: ['24-11-2021', '26-11-2021', '29-11-2021'],
      benchmark: [1.00267, 0.980307, 0.992332, 0.973024],
      portfolio: [1.0028, 0.9793, 0.99247],
   },
   ts_drawdowns: {
      date: ['24-11-2021', '26-11-2021', '29-11-2021'],
      benchmark: [0.0, -0.022303, -0.01031],
      portfolio: [0.0, -0.023434, -0.010301],
   },
}
const analysis = {
   fundamentals: {
      total_assets: 4,
      total_positions: 4,
      pe: 73.739873,
      pb: 28.025954,
      dividend_yield: 0.517303,
   },
   risk: {
      sharpe: -1.002506,
      volatility: 0.068686,
      beta: 0.315348,
      alpha: -0.000072,
   },
   returns: {
      returns_1M: -0.0148,
      returns_3M: -0.0103,
      returns_6M: null,
      CAGR: -0.0575,
   },
   capital_gains: {
      outstanding_balance: 1310000,
      unrealized_cap_losses: 101000,
      unrealized_cap_gains: 11234,
      realized_cap_losses: 0,
      realized_cap_gains: 0,
      taxes_cost: 0,
      taxes_rate: 0.4,
   },
   sectors: {
      sector: ['Cash', 'Consumer Discretionary', 'Information Technology'],
      percentage: [0.763359, 0.007634, 0.229008],
   },
   factors: {
      factor: [
         'style',
         'size',
         'yield',
         'quality',
         'volatility',
         'momentum',
         'liquidity',
      ],
      percentage: [
         0.513052, 0.618264, 0.59449, 0.5672, 0.432629, 0.529742, 0.465484,
      ],
   },
   segment_matrix: {
      style: ['Val', 'Val', 'Val', 'Bld', 'Bld', 'Bld', 'Gwth', 'Gwth', 'Gwth'],
      size: [
         'Large',
         'Mid',
         'Small',
         'Large',
         'Mid',
         'Small',
         'Large',
         'Mid',
         'Small',
      ],
      weight: [0, 0, 0, 19.083969, 0, 0, 4.580153, 0, 0],
   },
   top_holdings: {
      ticker: ['$CASH', 'MSFT', 'AAPL', 'TSLA'],
      weight: [0.763359, 0.19084, 0.038168, 0.007634],
      outstanding_balance: [1000000, 250000, 50000, 10000],
      unrealized_capital_gains: [0, -65000, 11234, -36000],
   },
   concentration: {
      ticker: ['$CASH', 'MSFT', 'AAPL', 'TSLA'],
      concentration: [76.335878, 95.419847, 99.236641, 100],
   },
   var: {
      quantile: [
         5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95,
      ],
      value: [
         0, 0, 0, 0, 0.010967, 0.021321, 0.030917, 0.040021, 0.04883, 0.0575,
         0.066169, 0.074978, 0.084083, 0.093678, 0.104032, 0.115563, 0.129003,
         0.145913, 0.170978,
      ],
   },
   risk_returns: {
      ticker: ['portfolio', 'benchmark'],
      returns: [-0.0575, -0.122369],
      volatility: [0.068686, 0.190293],
      sharpe: [0.837133, 0.643053],
   },
}

describe('test in Backtest tab', () => {
   const mockStore = configureStore()
   const store = mockStore(exampleReduxState)

   test('match snapshot', () => {
      const wrapper = mount(
         <Provider store={store}>
            <BrowserRouter>
               <TotalOverview backtest={backtest} analysis={analysis} />
            </BrowserRouter>
         </Provider>
      )
      expect(wrapper).toMatchSnapshot()
   })

   it('click on information icons', () => {
      const component = mount(
         <Provider store={store}>
            <BrowserRouter>
               <TotalOverview backtest={backtest} analysis={analysis} />
            </BrowserRouter>
         </Provider>
      )
      component
         .find({ 'data-testid': 'info-summary-overview-market_cap' })
         .simulate('click')
      component.find({ 'data-testid': 'info-summary-overview-pe' }).simulate('click')
      component
         .find({ 'data-testid': 'info-summary-overview-yield' })
         .simulate('click')
   })
})
