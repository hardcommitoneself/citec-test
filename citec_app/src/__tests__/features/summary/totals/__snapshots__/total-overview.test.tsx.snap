// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in Backtest tab match snapshot 1`] = `ReactWrapper {}`;
